audformat/core/table.py

```diff
@@ -28,7 +28,439 @@
+                if os.path.exists(csv_file):
+                    self._load_csv(csv_file)
+                    self._save_pickled(pkl_file)
+                else:
+                    raise ex
+        else:
+            self._load_csv(csv_file)
+
+    def save(
+            self,
+            path: str,
+            *,
+            storage_format: str = define.TableStorageFormat.CSV,
+            update_other_formats: bool = True,
+    ):
+        r"""Save table data to disk.
+
+        Existing files will be overwritten.
+
+        Args:
+            path: file path without extension
+            storage_format: storage format of table.
+                See :class:`audformat.define.TableStorageFormat`
+                for available formats
+            update_other_formats: if ``True`` it will not only save
+                to the given ``storage_format``,
+                but update all files stored in other storage formats as well
+
+        """
+        path = audeer.path(path)
+        define.TableStorageFormat.assert_has_attribute_value(storage_format)
+
+        pickle_file = path + f'.{define.TableStorageFormat.PICKLE}'
+        csv_file = path + f'.{define.TableStorageFormat.CSV}'
+
+        # Make sure the CSV file is always written first
+        # as it is expected to be older by load()
+        if storage_format == define.TableStorageFormat.PICKLE:
+            if update_other_formats and os.path.exists(csv_file):
+                self._save_csv(csv_file)
+            self._save_pickled(pickle_file)
+
+        if storage_format == define.TableStorageFormat.CSV:
+            self._save_csv(csv_file)
+            if update_other_formats and os.path.exists(pickle_file):
+                self._save_pickled(pickle_file)
+
+    def set(
+            self,
+            values: typing.Union[
+                typing.Dict[str, Values],
+                pd.DataFrame,
+            ],
+            *,
+            index: pd.Index = None,
+    ):
+        r"""Set labels.
+
+        By default, all labels of the table are replaced,
+        use ``index`` to select a subset.
+        If a column is assigned to a :class:`Scheme`
+        values have to match its ``dtype``.
+
+        Examples are provided with the
+        :ref:`table specifications <data-tables:Tables>`.
+
+        Args:
+            values: dictionary of values with ``column_id`` as key
+            index: index
+
+        Raises:
+            ValueError: if values do not match scheme
+
+        """
+        for idx, data in values.items():
+            self.columns[idx].set(data, index=index)
+
+    def _get_by_index(
+            self,
+            index: pd.Index,
+    ) -> (pd.DataFrame, bool):  # pragma: no cover
+        raise NotImplementedError()
+
+    def _index_levels_and_converters(self) -> typing.Tuple[
+            typing.Sequence[str],
+            typing.Dict[str, typing.Callable],
+    ]:  # pragma: no cover
+        raise NotImplementedError()
```

Comment:
Here also some comments would be nice, e.g.

```suggestion
        # Executed when loading table from CSV file.
        # Returns: `index_levels, index_converters`
        raise NotImplementedError()
```